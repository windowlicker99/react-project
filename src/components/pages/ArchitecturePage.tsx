import React, { useState, FC, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { ArchitectureControls } from '@components/page-contents/ArchitectureControls';
import { Btn } from '@components/common/form-controls/Btn';
import { ArchitectureScheme } from '@components/page-contents/ArchitectureScheme';
import { ModalCenter } from '@components/modules/ModalCenter';
import { AssignRA } from '@components/page-contents/AssignRA';
import { ArchitectureAside } from '@components/common/ArchitectureAside';
import { ArchitectureDetailsContent } from '@components/common/aside-content/ArchitectureDetailsContent';
import { EAside } from '@interfaces/enums';
import { selectCurrentVehicle } from '@store/selectors/vehicle';
import { getCurrentVehicle } from '@utils/getCurrentVehicle';
import { IGatewayResponse, IVehicle } from '@interfaces/interfaces';
import { DomainDetails } from '@components/page-contents/DomainDetails';
import { ECUDetails } from '@components/page-contents/ECUDetails';
import { getVehicleGateway } from '@utils/getVehicleGateway';
import { setCurrentElement } from '@store/slices/elementSlice';
import { InitiateRiskAssesment } from '@components/page-contents/InitiateRiskAssesment';
import { currentElementSelector } from '@store/selectors/element';
import { getSchemeData } from '@utils/getSchemeData';
import { IArchitectureDomain } from '@interfaces/ArchitectureScheme/interfaces';
import { Preloader } from '@components/common/Preloader';
import { setHeader } from '@/store/slices/headerSlice';

export const ArchitecturePage: FC = () => {
  const dispatch = useDispatch();
  const [vehicle, setVehicle] = useState<IVehicle>(null);
  const [schemeData, setSchemeData] = useState<IArchitectureDomain[]>(null);
  const [isOpenDomainModal, setIsOpenDomainModal] = useState<boolean>(false);
  const [isOpenECUModal, setIsOpenECUModal] = useState<boolean>(false);
  const [isOpenHistoryView, setIsOpenHistoryView] = useState<boolean>(false);
  const [isOpenRAModal, setIsOpenRAModal] = useState<boolean>(false);
  const [isOpenInitiateView, setIsOpenInitiateView] = useState<boolean>(false);
  const [gateway, setGateway] = useState<IGatewayResponse>(null);
  const [loading, setLoading] = useState(true);

  const currentElement = useSelector(currentElementSelector);

  const currentVehicle = useSelector(selectCurrentVehicle);

  const onOpenDomainModalClick = () => {
    setIsOpenDomainModal(true);
  };

  const onCloseDomainModalClick = () => {
    setIsOpenDomainModal(false);
  };

  const onOpenECUModalClick = () => {
    setIsOpenECUModal(true);
  };

  const onCloseECUModalClick = () => {
    setIsOpenECUModal(false);
  };

  const onOpenHistoryView = () => {
    setIsOpenHistoryView(true);
  };

  const onCloseHistoryView = () => {
    setIsOpenHistoryView(false);
  };

  const onOpenInitiateView = () => {
    setIsOpenInitiateView(true);
  };

  const onOpenRAModal = () => {
    setIsOpenRAModal(true);
  };
  const onCloseRAModal = () => {
    setIsOpenRAModal(false);
  };

  const onApplyModal = () => {
    console.log('apply'); // temp
  };

  const getVehicle = async () => {
    if (!currentVehicle) {
      setVehicle(null);
      return;
    }
    const currentVehicleData = await getCurrentVehicle(currentVehicle);
    setVehicle(currentVehicleData);
  };

  const onCloseInitiateView = () => {
    setIsOpenInitiateView(false);
    getVehicle();
  };

  const getController = async (vehicleId: string) => {
    try {
      setLoading(true);
      const { gateway, type } = await getVehicleGateway(vehicleId);
      setGateway({ gateway, type });
      dispatch(setCurrentElement({ type: EAside[type], id: gateway?.id }));
      setLoading(false);
    } catch (error) {
      dispatch(setCurrentElement(null));
    }
  };

  const getArchitectureSchemeData = async (id: string) => {
    const newSchemeData = await getSchemeData(id);
    setSchemeData(newSchemeData);
    setLoading(false);
  };

  useEffect(() => {
    if (!isOpenDomainModal || isOpenECUModal) {
      getArchitectureSchemeData(currentVehicle);
    }
  }, [currentVehicle, isOpenDomainModal, isOpenECUModal]);

  useEffect(() => {
    getVehicle();
  }, [currentVehicle]);

  useEffect(() => {
    if (currentVehicle && !gateway && !isOpenECUModal) {
      getController(currentVehicle);
    }
  }, [currentVehicle, isOpenECUModal]);

  useEffect(() => {
    if (vehicle) {
      const subtitles = [];
      if (vehicle.model) {
        subtitles.push(vehicle.model);
      }

      if (vehicle.brand?.name) {
        subtitles.push(vehicle?.brand?.name);
      }
      dispatch(setHeader(subtitles));
    }
  }, [vehicle]);

  return (
    <div className="content" id="architecture">
      <ArchitectureControls
        vehicle={vehicle}
        onOpenDomainModalClick={onOpenDomainModalClick}
        onOpenHistoryView={onOpenHistoryView}
      />
      <div className="page-content">
        {loading ? (
          <Preloader />
        ) : (
          <>
            <div className="page-thumb">
              <div className="architecture-scheme" id="architectureScheme">
                <ArchitectureScheme
                  gatewayInfo={gateway}
                  schemeData={schemeData}
                  onECUCreateClick={onOpenECUModalClick}
                />
              </div>
              <div className="architecture-controls">
                <div className="page-controls-main">
                  <div className="btn-box">
                    <Btn btnStatus="btn-control" btnIcon="icon-grid" />
                    <Btn btnStatus="btn-control" btnIcon="icon-list" />
                  </div>
                  <div className="btn-box">
                    <Btn btnStatus="btn-control" btnIcon="icon-maximize" />
                    <div className="btn-box-separator" />
                    <Btn btnStatus="btn-control" btnIcon="icon-minus" />
                  </div>
                </div>
                <Btn btnStatus="btn-control" btnIcon="icon-chevrons-right" />
              </div>
            </div>
            <ArchitectureAside currentElement={currentElement} asideStatus="large">
              <ArchitectureDetailsContent
                onAssign={onOpenRAModal}
                element={currentElement}
                isAssignModalOpen={isOpenRAModal}
                onOpenInitiateView={onOpenInitiateView}
                isInitiateModalOpen={isOpenInitiateView}
              />
            </ArchitectureAside>
          </>
        )}
      </div>
      <DomainDetails
        vehicle={vehicle}
        isOpenDomainModal={isOpenDomainModal}
        onCloseDomainModalClick={onCloseDomainModalClick}
      />
      <ECUDetails vehicle={vehicle} isOpen={isOpenECUModal} onClose={onCloseECUModalClick} />
      <AssignRA element={currentElement} isOpen={isOpenRAModal} onClose={onCloseRAModal} />
      {/* Add content to modals */}
      <ModalCenter title="History" isOpen={isOpenHistoryView} onApply={onApplyModal} onClose={onCloseHistoryView} />
      <InitiateRiskAssesment
        element={currentElement}
        vehicle={vehicle}
        isOpenInitiateView={isOpenInitiateView}
        onCloseInitiateView={onCloseInitiateView}
      />
    </div>
  );
};
