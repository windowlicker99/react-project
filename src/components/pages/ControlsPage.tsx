import React, { FC, useState } from 'react';
import { Tab, TabPanel, TabList, Tabs } from 'react-tabs';
import { useSelector } from 'react-redux';
import { PageSearchbar } from '@components/page-contents/PageSearchbar';
import { Table } from '@components/page-contents/Table';
import { Footer } from '@components/modules/Footer';
import classNames from 'classnames';
import { Preloader } from '@components/common/Preloader';
import { Btn } from '@components/common/form-controls/Btn';
import { selectColumnsWithDropdownList } from '@/store/selectors/columns';
import { controlsTabs } from '@/constants/tabsLists';

export const ControlsPage: FC = () => {
  const [searchText, setSearchText] = useState<string>('');

  const [loading, setLoading] = useState(true);

  const customizedColumns = useSelector(selectColumnsWithDropdownList);

  const [selectedFindingsTabIndex, setSelectedFindingsTabIndex] = useState(0);

  return (
    <Tabs
      className="content"
      id="controls"
      onSelect={setSelectedFindingsTabIndex}
      selectedIndex={selectedFindingsTabIndex}
      selectedTabPanelClassName="box-table box-not-radius"
    >
      <div className="box page-controls">
        <PageSearchbar
          searchText={searchText}
          onSearch={setSearchText}
          onCustomizeColumnsClick={(): null => null} // temp
          btnText="Create Control"
          btnIcon="icon-plus"
          showFilterBtn={false}
        />
      </div>

      <TabList className="tabs-box">
        {controlsTabs.map((tab, i) => (
          <Tab key={tab.name}>
            <Btn
              btnStatus={classNames('btn-stretch', selectedFindingsTabIndex === i ? 'btn-lg-primary' : 'btn-lg-info')}
              btnText={tab.name}
            />
          </Tab>
        ))}
      </TabList>

      {controlsTabs.map((tab) => (
        <TabPanel key={tab.name}>
          <Table
            changeLoadingState={setLoading}
            columns={customizedColumns}
            setSearchText={setSearchText}
            searchText={searchText}
            tableType={tab.type}
            withoutPagination
          />
        </TabPanel>
      ))}
      {loading ? <Preloader /> : null}
      <Footer />
    </Tabs>
  );
};
