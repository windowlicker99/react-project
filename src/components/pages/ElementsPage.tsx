import React, { FC, useState } from 'react';
import { Tab, TabPanel, TabList, Tabs } from 'react-tabs';
import { useDispatch, useSelector } from 'react-redux';
import { useFormik } from 'formik';
import { PageFilter } from '@components/page-contents/PageFilter';
import { PageSearchbar } from '@components/page-contents/PageSearchbar';
import { Table } from '@components/page-contents/Table';
import { elementsTabsList } from '@constants/tabsLists';
import { ModalCenter } from '@components/modules/ModalCenter';
import { CustomiseModalContent } from '@components/modal-content/CustomiseModalContent';
import { selectColumnsInitialValues, selectColumnsList, selectColumnsFieldsList } from '@store/selectors/columns';
import { updateDisplayedColumns } from '@store/slices/columnsSlice';
import { Preloader } from '@components/common/Preloader';
import { Footer } from '@components/modules/Footer';

export const ElementsPage: FC = () => {
  const dispatch = useDispatch();
  const [searchText, setSearchText] = useState<string>('');
  const [loading, setLoading] = useState(true);

  const [isOpenCustomizeModal, setIsOpenCustomizeColumnsModal] = useState<boolean>(false);
  const initialValues = useSelector(selectColumnsInitialValues);
  const columnsList = useSelector(selectColumnsList);
  const columnsFields = useSelector(selectColumnsFieldsList);

  const formik = useFormik({
    initialValues,
    enableReinitialize: true,
    onSubmit: (values) => {
      dispatch(updateDisplayedColumns({ displayColumnsList: values }));
    },
  });

  const onOpenCustomizeModal = () => {
    setIsOpenCustomizeColumnsModal(true);
  };

  const onApplyCustomizeModal = () => {
    formik.submitForm();
    setIsOpenCustomizeColumnsModal(false);
  };

  const onCloseCustomizeModal = () => {
    formik.resetForm();
    setIsOpenCustomizeColumnsModal(false);
  };

  return (
    <>
      <Tabs className="content" id="elements" selectedTabPanelClassName="box-table elements-content">
        <div className="box page-controls">
          <PageSearchbar
            searchText={searchText}
            onSearch={setSearchText}
            onCustomizeColumnsClick={onOpenCustomizeModal}
            btnText="Customize Columns"
            btnIcon="icon-columns"
          />
          <PageFilter />
          <TabList className="page-tabs">
            {elementsTabsList.map((tab) => (
              <Tab className="page-tabs-item" selectedClassName="page-tabs-item active" key={tab.name}>
                {tab.name}
              </Tab>
            ))}
          </TabList>
        </div>

        {elementsTabsList.map((tab) => (
          <TabPanel key={tab.name}>
            <Table
              columns={columnsList}
              changeLoadingState={setLoading}
              setSearchText={setSearchText}
              searchText={searchText}
              tableType={tab.type}
              showHeader
            />
          </TabPanel>
        ))}
        <Footer />
      </Tabs>

      {loading ? <Preloader /> : null}

      <ModalCenter
        title="Customize columns"
        isOpen={isOpenCustomizeModal}
        onApply={onApplyCustomizeModal}
        onClose={onCloseCustomizeModal}
      >
        <CustomiseModalContent columns={columnsFields} {...formik} />
      </ModalCenter>
    </>
  );
};
