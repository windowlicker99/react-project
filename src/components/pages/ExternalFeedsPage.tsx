import React, { FC, useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
import { Heading } from '@components/common/Heading';
import { Table } from '@components/page-contents/Table';
import { Aside } from '@components/common/Aside';
import { CardAlert } from '@components/common/CardAlert';
import { ETableTypes, IAlert } from '@interfaces/interfaces';
import { selectColumnsWithExpandAllDropdownList } from '@store/selectors/columns';
import { getData } from '@utils/getElements';
import { API_PATHS } from '@constants/paths';
import { Preloader } from '@components/common/Preloader';

export const ExternalFeedsPage: FC = () => {
  const [alerts, setAlerts] = useState<IAlert[]>([]);
  const [loading, setLoading] = useState(true);

  const columns = useSelector(selectColumnsWithExpandAllDropdownList);

  const getAlerts = async () => {
    const newAlerts: IAlert[] = await getData(API_PATHS.alerts);
    setAlerts(newAlerts);
  };

  useEffect(() => {
    getAlerts();
  }, []);

  return (
    <Tabs className="page-holder" id="feeds">
      <div className="aside-holder">
        {!loading ? <Heading headingText="Car Alerts" headingNumber={`(${alerts.length})`} /> : null}
        <Aside asideStatus="small transparent">
          <TabList>
            {alerts.map((alert) => (
              <Tab className="tab-item" selectedClassName="active" key={`tab-${alert.name}`}>
                <CardAlert alert={alert} />
              </Tab>
            ))}
          </TabList>
        </Aside>
      </div>

      {alerts.map((alert) => (
        <TabPanel className="tab-content" key={`tabpanel-${alert.name}`}>
          <Heading headingText="Vehicle Types" headingNumber={`(${alert?.vehicleIds.length})`} subtitle={alert.name} />
          <div className="box-table">
            <Table
              changeLoadingState={setLoading}
              tableType={ETableTypes.feeds}
              columns={columns}
              params={{ alertId: alert.id }}
              showHeader
            />
          </div>
        </TabPanel>
      ))}
      {loading ? <Preloader /> : null}
    </Tabs>
  );
};
