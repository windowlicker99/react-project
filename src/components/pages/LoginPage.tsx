import React, { FC } from 'react';
import { LoginForm } from '@components/page-contents/LoginForm';
import { Logo } from '@components/common/Logo';

export const LoginPage: FC = () => (
  <div className="page-container is-open">
    <div className="login-image">
      <Logo />
    </div>
    <LoginForm />
  </div>
);
