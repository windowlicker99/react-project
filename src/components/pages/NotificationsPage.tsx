import React, { FC } from 'react';
import { PageControls } from '@components/page-contents/PageControls';
import { NotificationsContent } from '@components/page-contents/NotificationsContent';

export const NotificationsPage: FC = () => (
  <div className="content" id="notifications">
    <PageControls />
    <NotificationsContent />
  </div>
);
