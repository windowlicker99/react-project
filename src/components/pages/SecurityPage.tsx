import React, { FC, useEffect, useState } from 'react';
import { Tab, TabList, TabPanel, Tabs } from 'react-tabs';
import { useDispatch, useSelector } from 'react-redux';
import { securityDevelopmentTabsList } from '@constants/tabsLists';
import { currentElementSelector } from '@/store/selectors/element';
import { EAside, ESecurityTypes } from '@/interfaces/enums';
import { getElement } from '@/utils/getElement';
import { setHeader } from '@/store/slices/headerSlice';
import { Btn } from '../common/form-controls/Btn';

export const SecurityPage: FC = () => {
  const dispatch = useDispatch();
  const [showButtons, setShowButtons] = useState(false);
  const currentElement = useSelector(currentElementSelector);

  const getECU = async () => {
    const ECU = await getElement(currentElement);
    dispatch(setHeader([ECU.name]));
  };

  const onSelect = (index: number) => {
    if (securityDevelopmentTabsList[index].type === ESecurityTypes.penetrationTests) {
      setShowButtons(true);
      return;
    }
    setShowButtons(false);
  };

  useEffect(() => {
    if (currentElement.type === EAside.ecu) {
      getECU();
    }
  }, []);

  return (
    <Tabs className="content" id="security" selectedTabPanelClassName="page-content" onSelect={onSelect}>
      <div className="box page-controls">
        <div className="page-controls-content tabs-bottom">
          <TabList className="page-tabs">
            {securityDevelopmentTabsList.map((tab) => (
              <Tab className="page-tabs-item" selectedClassName="page-tabs-item active" key={tab.name}>
                {tab.name}
              </Tab>
            ))}
          </TabList>
          {showButtons && (
            <div className="page-controls-item mb-20">
              <Btn btnStatus="btn-lg-primary" btnIcon="icon-plus" btnText="Add Report" />
              <Btn btnStatus="btn-lg-outline ml-15" btnIcon="" btnText="Initiate PT" />
              <Btn btnStatus="btn-xs-primary ml-15" btnIcon="icon-user-plus" />
            </div>
          )}
        </div>
      </div>

      {securityDevelopmentTabsList.map((tab) => (
        <TabPanel key={tab.name}>{tab.component}</TabPanel>
      ))}
    </Tabs>
  );
};
