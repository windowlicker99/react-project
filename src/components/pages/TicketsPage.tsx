import React, { FC } from 'react';
import { PageControls } from '@components/page-contents/PageControls';
import { TicketsContent } from '@components/page-contents/TicketsContent';

export const TicketsPage: FC = () => (
  <div className="content" id="tickets">
    <PageControls />
    <TicketsContent />
  </div>
);
