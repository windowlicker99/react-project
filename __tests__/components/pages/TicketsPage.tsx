import React from 'react';
import { render } from '@testing-library/react';
import { TicketsPage } from '@components/pages/TicketsPage';

describe('TicketsPage tests', () => {
  it('Should render', () => {
    const { container } = render(<TicketsPage />);
    expect(container.getElementsByClassName('content').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(<TicketsPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
