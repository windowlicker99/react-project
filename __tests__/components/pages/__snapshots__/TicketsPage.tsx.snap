// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketsPage tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
    id="tickets"
  >
    <div
      class="box page-controls"
    >
      <div
        class="page-controls-content flex-end"
      >
        <div
          class="page-controls-item"
        >
          <div
            class="mr-15"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label search-label"
                for="exampleFormControlInput1"
              >
                What are you looking for?
              </label>
              <div
                class="search-group"
              >
                <div
                  class="search-icon"
                >
                  <span
                    class="icon-loupe"
                  />
                </div>
                <input
                  class="form-control search-input"
                  id="exampleFormControlInput1"
                  placeholder="Search"
                  type="email"
                  value=""
                />
                <button
                  class="search-btn"
                  type="button"
                >
                  <span
                    class="icon-cross"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="mr-15"
          >
            <button
              class="btn-xs-primary"
              type="button"
            >
              <span
                class="icon-filter-blue"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="page-filter"
      >
        <div
          class="page-filter-inner"
        />
        <div
          class="page-filter-btn"
        >
          <button
            class="btn btn-xs-secondary mr-15"
            type="button"
          >
            <span
              class="btn-icon icon-close"
            />
            <span
              class="btn-text"
            />
          </button>
          <button
            class="btn btn-lg-outline"
            type="button"
          >
            <span
              class="btn-icon "
            />
            <span
              class="btn-text"
            >
              Apply
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="box-wrap tickets-content"
    >
      <div
        class="box-col open"
      >
        <div
          class="box-col-header"
        >
          <div
            class="heading"
          >
            <div
              class="heading-info"
            >
              <div
                class="badge-round sky-blue"
              />
              <span
                class="heading-title"
              >
                Open
              </span>
              <span
                class="heading-data"
              >
                (0) 
              </span>
            </div>
          </div>
        </div>
        <div
          class="box-col-content"
        />
      </div>
      <div
        class="box-col progress"
      >
        <div
          class="box-col-header"
        >
          <div
            class="heading"
          >
            <div
              class="heading-info"
            >
              <div
                class="badge-round yellow-dark"
              />
              <span
                class="heading-title"
              >
                In Progress
              </span>
              <span
                class="heading-data"
              >
                (0) 
              </span>
            </div>
          </div>
        </div>
        <div
          class="box-col-content"
        />
      </div>
      <div
        class="box-col closed"
      >
        <div
          class="box-col-header"
        >
          <div
            class="heading"
          >
            <div
              class="heading-info"
            >
              <div
                class="badge-round green"
                style="background-color: green;"
              />
              <span
                class="heading-title"
              >
                Close
              </span>
              <span
                class="heading-data"
              >
                (0) 
              </span>
            </div>
          </div>
        </div>
        <div
          class="box-col-content"
        />
      </div>
      <div
        class="box-col deleted"
      >
        <div
          class="box-col-header"
        >
          <div
            class="heading"
          >
            <div
              class="heading-info"
            >
              <div
                class="badge-round red-light"
              />
              <span
                class="heading-title"
              >
                Deleted
              </span>
              <span
                class="heading-data"
              >
                (0) 
              </span>
            </div>
          </div>
        </div>
        <div
          class="box-col-content"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
